transmogrifier/sources/dspace_dim.py

```diff
@@ -0,0 +1,276 @@
+        kwargs = {
+            "source": source_name,
+            "source_link": source_base_url + source_record_id,
+            "timdex_record_id": f"{source}:{source_record_id.replace('/', '-')}",
+            "title": main_title[0].string,
+        }
+
+        # Optional fields in TIMDEX
+        # alternate_titles, uses full title list retrieved for main title field
+        for alternate_title in [
+            t
+            for t in all_titles
+            if "qualifier" in t.attrs and t.attrs["qualifier"] == "alternative"
+        ]:
+            a = AlternateTitle(
+                value=alternate_title.string,
+                kind="AlternativeTitle",
+            )
+            kwargs.setdefault("alternate_titles", []).append(a)
+
+        # contributors
+        citation_creators = []
+        for creator in [f for f in all_fields if f.attrs["element"] == "creator"]:
+            creator_name = creator.string
+            citation_creators.append(creator_name)
+            c = Contributor(
+                value=creator_name,
+                kind="Creator",
+            )
+            kwargs.setdefault("contributors", []).append(c)
+
+        for contributor in [
+            f for f in all_fields if f.attrs["element"] == "contributor"
+        ]:
+            c = Contributor(
+                value=contributor.string,
+            )
+            if "qualifier" in contributor.attrs:
+                c.kind = contributor.attrs["qualifier"]
+            kwargs.setdefault("contributors", []).append(c)
+
+        # dates
+        date_issued = ""
+        for date in [f for f in all_fields if f.attrs["element"] == "date"]:
+            d = Date(value=date.string)
+            if "qualifier" in date.attrs:
+                d.kind = date.attrs["qualifier"]
+            if date.attrs["qualifier"] == "issued":
+                date_issued = date.string
+            kwargs.setdefault("dates", []).append(d)
+
+        # file_formats
+        for file_format in [
+            f
+            for f in all_fields
+            if f.attrs["element"] == "format"
+            and "qualifier" in f.attrs
+            and f.attrs["qualifier"] == "mimetype"
+        ]:
+            kwargs.setdefault("file_formats", []).append(file_format.string)
+
+        # format
+        kwargs["format"] = "electronic resource"
+
+        # funding_information
+        for funding_reference in [
+            f
+            for f in all_fields
+            if f.attrs["element"] == "description"
+            and "qualifier" in f.attrs
+            and f.attrs["qualifier"] == "sponsorship"
+        ]:
+            fr = Funder(
+                funder_name=funding_reference.string,
+            )
+            kwargs.setdefault("funding_information", []).append(fr)
+
+        # identifiers
+        identifiers = [f for f in all_fields if f.attrs["element"] == "identifier"]
+        for identifier in identifiers:
+            if "qualifier" not in identifier.attrs:
```

Comment:
This identifiers section can be simplified by just excluding citation identifiers and then using the dict.get() method to set the kind, e.g.:
```
if identifier.get("qualifier") != "citation":
    i = Identifier(
        value=identifier.string,
        kind=identifier.get("qualifier"),
    )
```
There are several other fields below where a similar simplification can be made.